pi-geofence/src/main/java/com/ibm/pi/geofence/PIGeofenceEvent.java

```diff
@@ -0,0 +1,143 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ibm.pi.geofence;
+
+import android.content.Intent;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ *
+ */
+public class PIGeofenceEvent {
+    /**
+     * The name of the action used as filter for geofence events delivered as intents.
+     */
+    public static final String ACTION_GEOFENCE_EVENT = "com.ibm.pi.geofence.action.GEOFENCE_EVENT";
+    private static final String GEOFENCES_KEY = "com.ibm.pi.geofence.geofences";
+    private static final String DELETED_GEOFENCES_KEY = "com.ibm.pi.geofence.deleted_geofences";
+    private static final String EVENT_TYPE_KEY = "com.ibm.pi.geofence.event_type";
+
+    /**
+     * The possible types of geofence events.
+     */
+    public enum Type {
+        /**
+         * Event type indicating entry within one or more geofences.
+         */
+        ENTER("enter"),
+        /**
+         * Event type indicating exit from one or more geofences.
+         */
+        EXIT("exit"),
+        /**
+         * Event type indicating results from a server synchronization request.
+         */
+        SERVER_SYNC("server_sync");
+
+        private String op;
+
+        Type(String op) {
+            this.op = op;
+        }
+
+        String operation() {
+            return op;
+        }
+    }
+
+    private final Type mEventType;
+    private final List<PIGeofence> mGeofences;
+    private final List<String> mDeletedGeofenceCodes;
+
+    private PIGeofenceEvent(final Type eventType, List<PIGeofence> geofences, final List<String> deletedGeofenceCodes) {
+        this.mEventType = eventType;
+        this.mGeofences = geofences;
+        this.mDeletedGeofenceCodes = deletedGeofenceCodes;
+    }
+
+    /**
+     * Get the list of the codes of the geofences that were deleted since the last synchronization.
+     * @return a list of codes for geofences that were deleted, or {@code null} if no geofence was deleted or
+     * {@code getEventType() != PIGeofenceEvent.Type.SERVER_SYNC}.
+     */
+    public List<String> getDeletedGeofenceCodes() {
+        return mDeletedGeofenceCodes;
+    }
+
+    /**
+     * Get the type of this event.
+     * @return one of the elements in the {@link Type} enum.
+     */
+    public Type getEventType() {
+        return mEventType;
+    }
+
+    /**
+     * Get the geofences this event relates to. The meaning of the result should can interpreted in two ways:
+     * <ul>
+     *     <li>if {@link #getEventType()} returns {@link Type#ENTER} or {@link Type#EXIT} then these are gefoences that were entered or exited</li>
+     *     <li>if {@link #getEventType()} returns {@link Type#SERVER_SYNC}, then these are the gefoences that were created or updated
+     *     since the last server synchronization</li>
+     * </ul>
+     * @return a list of {@link PIGeofence} objects, possibly {@code null}.
```

Comment:
The meaning of the result can be interpreted in two ways: *

<li>if {@link #getEventType()} returns {@link Type#ENTER} or {@link Type#EXIT}, then these are geofences that were entered or exited</li>*

geofences* (in second bullet as well)